csrc/codegen.cpp

```diff
@@ -295,8 +295,36 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
       NVF_ERROR(
           num_threads_per_cta.has_value(),
           "__launch_bounds__ must be set for register sharing warp specialization");
-      code_ << "__launch_bounds__(/*MAX_THREADS_PER_BLOCK=*/"
-            << num_threads_per_cta.value() << ") ";
+
+      int64_t initial_reg_count =
+          getRegPerThreadGivenThreadsPerSM(num_threads_per_cta.value());
+
+      NVF_ERROR(
+          kernel_->hasManaged("decreased_register_count") &&
+              kernel_->hasManaged("increased_register_count"),
+          "Decreased and increased register count must be set for register sharing warp specialization");
+
+      int64_t decreased_reg_count =
```

Comment:
Can this check occur when compiling the fusion like a vectorization check? We can have multiple circular buffered loops. You can look up the register count through the kernel summary.

```
      int64_t prefetch = kernel_->summary()
                             .circular_buffer_info
                             .getCircularBufferOptionsFor(loop->iter_domain())
                             .prefetch;
```